### PR TITLE
Add setting to set the Date Format 

### PR DESCRIPTION
**Pull request by fluorine**
Feature to select the date format of the transactions, discusses in issue #450 

An option to select the Date Format has been included in the configuration menu, just after the setting for language.

Date formats added:
-  MM/DD/YYYY - Used mainly in the US
-  DD/MM/YYYY - Common in Europe
-  YYYY/MM/DD - Used in Asia

### CODE REVIEW

File: client/app/src/accounts/account.controller.js [hunk 1]
```diff
@@ -402,6 +434,26 @@
       storageService.set('language', self.language)
       gettextCatalog.setCurrentLanguage(self.language)
     }
+		
+    self.selectAllDateFormats = function () {
+      return dateFormats
+    }
+	
+    self.setDateFormat = function () {
+      function getdateformat (value) {
```
Comment (spresnal) [inline client/app/src/accounts/account.controller.js]: Did you intend to give this method an all lowercase name?

File: client/app/src/accounts/account.controller.js [hunk 2]
```diff
@@ -38,6 +38,25 @@
         return username
       }
     }])
+    .filter('formattedDate', ['storageService', '$filter', function (storageService, dateFilter) {
```
Comment (spresnal) [inline client/app/src/accounts/account.controller.js]: Unit tests for this formattedDate filter would be a good idea

File: client/app/src/accounts/account.controller.js [hunk 3]
```diff
@@ -99,6 +118,12 @@
       es_419: gettextCatalog.getString('Spanish'),
       sv: gettextCatalog.getString('Swedish')
     }
+	
+	var dateFormats = {
```
Comment (spresnal) [inline client/app/src/accounts/account.controller.js]: This should be a constant

File: client/app/src/accounts/account.controller.js (hunk 2 above)
Comment (fluorine) [reply client/app/src/accounts/account.controller.js]: I'm new in Angular and JS in general, but I'll learn how to create that test and then I'll submit it.

Comment (j-a-m-l) [reply client/app/src/accounts/account.controller.js]: Thanks for your effort!

**Comment by fluorine** [comment]
I have some problems. I want to do these things:
 - Fix the refresh for the dates after format is selected - It only refreshes once and I don't know why. Can someone give me a hint?
 - I'm trying to add a test to the filter. What testing tool we use? Can any of you provide me any reference?

It is my intention to contribute to this project. Thanks a lot!

**Comment by j-a-m-l** [comment]
* I'll ensure that the filter is triggered again.
* We use Mocha, Chai and Sinon for tests. In the `test` folder you can see some examples.

**Comment by fluorine** [comment]
I'm sorry I have not pushed the tests. I have written them, but I have not run them because I have problems with sinon-chai. This is the problem:

> C:\Users\Joseph\Desktop\ArkProject\ark-desktop>npm install sinon-chai
> npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@^1.0.0 (node_modules\chokidar\node_modules\fsevents):
> npm WARN optional SKIPPING OPTIONAL DEPENDENCY: sha512-WIr7iDkdmdbxu/Gh6eKEZJL6KPE74/5MEsf2whTOFNxbIoIixogroLdKYqB6FDav4Wavh/lZdzzd3b2KxIXC5Q== integrity checksum failed when using sha512: wanted sha512-WIr7iDkdmdbxu/Gh6eKEZJL6KPE74/5MEsf2whTOFNxbIoIixogroLdKYqB6FDav4Wavh/lZdzzd3b2KxIXC5Q== but got sha512-4lGswPDk0jk9f4+fPbGBnfm4bJHFPGsBcKNmXDNCaGQFaMRWZTPPF1osPNtFPCQ3fAlBKocc9vX1gHlJP4tuug==. (1374775 bytes)
> npm WARN optional SKIPPING OPTIONAL DEPENDENCY: 7zip-bin-linux@1.2.0 (node_modules\7zip-bin-linux):
> npm WARN optional SKIPPING OPTIONAL DEPENDENCY: sha512-umB98LN18XBGKPw4EKET2zPDqVhEU1mxXA1Gx0BM+DoBt4hnlZPNkpSMNzmuNbQshi9SzLhqlTAyKcAgNrbV3Q== integrity checksum failed when using sha512: wanted sha512-umB98LN18XBGKPw4EKET2zPDqVhEU1mxXA1Gx0BM+DoBt4hnlZPNkpSMNzmuNbQshi9SzLhqlTAyKcAgNrbV3Q== but got sha512-7w9MerqIF9ufugN+YxiixvCGoxFVpOhu16Im7QvzPCM5GEFLjbUgA7cVAd2eife7l4Ben05GzuKAQMJyAr9GIQ==. (29755 bytes)
> 
> npm ERR! code EINTEGRITY
> npm ERR! sha512-9stIF1utB0ywNHNT7RgiXbdmen8QDCRsrTjw+G9TgKt1Yexjiv8TOWZ6WHsTPz57Yky3DIswZvEqX8fpuHNDtQ== integrity checksum failed when using sha512: wanted sha512-9stIF1utB0ywNHNT7RgiXbdmen8QDCRsrTjw+G9TgKt1Yexjiv8TOWZ6WHsTPz57Yky3DIswZvEqX8fpuHNDtQ== but got sha512-pEK1ZUxxWLHXAu8Exm/y6iTtlN9DfOwG9lfaoNhXxyAjLzDl8vZwidaqj3+vahTgAlhr2CIYBXn7EAqEVTIX5Q==. (2341 bytes)
> 
> npm ERR! A complete log of this run can be found in:
> npm ERR!     C:\Users\Joseph\AppData\Roaming\npm-cache\_logs\2017-12-20T04_15_02_721Z-debug.log

**Comment by fluorine** [comment]
I had to pull #478 changes so I could run the Unit tests in windows.

**Comment by fluorine** [comment]
The online test run was failing because the time and day varies, since the format outputs the local date time (which is different in my machine). 

Therefore, the test was fixed to only test the date part and the order, which is what the filter does.

**Comment by Nasicus** [comment]
@fluorine 
Glad I could help!

Now we (the ark team :D ) probably should my PR first or else we have a "little" mess.

BTW: If my PR gets merged first I suggest you do the following after (for a clean history):
- Pull master
- Git reset mixed your branch to the master branch
- Add one or two commits for your entire logic
- Force push

If you'd like to do this and need any help just write me in gitter or so :)

**Comment by fluorine** [comment]
@Nasicus You are right. I will be watching your issue and I'll do what you suggested for a clearer history. Thanks again!

**Comment by alexbarnsley** [comment]
@fluorine Was you waiting on #478 to be merged first?

Also, travis seems to be failing, probably because of these:

image

There's also a conflict. Fix these and we will give it a test 👌  (sorry - a bit late!)

File: client/app/src/accounts/account.controller.js [hunk 4]
```diff
@@ -434,6 +481,198 @@
...
+        }
+      }
+
+      self.dateFormat = getDateFormat(this.selectedDateFormat)
+      storageService.set('dateFormat', self.dateFormat)
+
+      refreshCurrentAccountView()
+    }
+
+    self.getMarketInfo = function (symbol) {
```
Comment (alexbarnsley) [inline client/app/src/accounts/account.controller.js]: I'm not too sure what's happened here @fluorine. Code is being added that was removed in PR https://github.com/ArkEcosystem/ark-desktop/pull/461

Comment (alexbarnsley) [reply client/app/src/accounts/account.controller.js]: Basically all Market stuff was removed because it's not used

**Comment by fluorine** [comment]
I hope the best for the project, but I'm not (trying to) contribute to the Ark Platform anymore.

I think any meaningful feedback from the team came too late (a month later) and therefore I prefer to focus on other projects, instead fixing old conflicts on code I don't remember anymore.

**Comment by alexbarnsley** [comment]
No worries @fluorine. We wasn't too sure if it was ready to be reviewed again as Travis was failing. Thanks again.

**Pull request closed without merging**